Lib/test/test_long.py

```diff
@@ -1349,6 +1349,11 @@ def test_shift_bool(self):
                 self.assertEqual(type(value << shift), int)
                 self.assertEqual(type(value >> shift), int)
 
+    def test_as_integer_ratio(self):
+        tests = [10, 0, -10, 1, 3]
```

Comment:
Why so much similar cases are needed?

Add tests for booleans and other int subclasses. Check types of numerator and denominator.